### PR TITLE
Add/warehouse

### PR DESCRIPTION
**Pull request by szm33**

### CODE REVIEW

File: src/main/java/com/zzpj/backend/api/v1/UserController.java [hunk 1]
```diff
@@ -20,31 +23,46 @@ public UserController(UserServiceLocal userService) {
...
 
     @PutMapping
-    public String edit(@RequestBody User user){
-        userService.editUser(user);
-        return "Success";
+    public ResponseEntity<String> edit(@RequestBody User user) {
+        try {
+            if(user.getId() != null && user.getId() < 0) throw new AppBaseException();
+            userService.editUser(user);
+        } catch (AppBaseException e) {
```
Comment (kcc112) [inline src/main/java/com/zzpj/backend/api/v1/UserController.java:58]: Simple `if (user.getId() != null && user.getId() < 0) `would be enough

File: src/main/java/com/zzpj/backend/services/implementations/AlcoholService.java [hunk 2]
```diff
@@ -22,7 +22,17 @@ public AlcoholService(AlcoholRepository alcoholRepository) {
 
     @Override
     public void addAlcohol(Alcohol alcohol) {
-        alcoholRepository.save(alcohol);
+      Optional<Alcohol> alcoholDB=  alcoholRepository.findAll()
+              .stream()
+              .filter(e-> e.getName().equals(alcohol.getName()))
+              .findFirst();
+      if(alcoholDB.isPresent()){
```
Comment (kcc112) [inline src/main/java/com/zzpj/backend/services/implementations/AlcoholService.java:29]: Please remember to add space between next time `if(alcoholDB.isPresent()) {`

**Review by kcc112 — DISMISSED** [review]
Few things to change

**Review by MaciejBartos — DISMISSED** [review]
lgtm

**Review by kcc112 — COMMENTED** [review]
Still, we need to change add alcohol and punches list endpoints

**Review by kcc112 — APPROVED** [review]

**Pull request merged**